### PR TITLE
Update the code to match the updated UI design in the design document

### PR DESCRIPTION
**Pull request by carbonXIII**
### Major UI changes
The current design document has a button that transitions between the timer and the task detail screen, but instead I have opted to use a tabbed layout so that we can replace what is currently the "Task" tab with a list of all tasks, where the user can change current tasks, modify the list, etc.

### Design document updates [TODO]
I updated the sequence diagrams, but these updated diagrams are not currently in the design document. The collaboration diagrams will need to be updated as well (and probably annotated with user story ID's for each class).

### Tests/Requirements
Some of the requirements no longer apply to the updated UI, so I've modified the tests to point to new requirements. All of the presenters have tests now though, and the existing test for the model classes have been updated.

There are still no tests for the view/UI side though. In the future we should probably use a testing framework like [Espresso](https://developer.android.com/training/testing/espresso) but manual tests will still work for now. These manual tests will need to be updated, but I have not done that either.

### Other minor changes
The Android API version was bumped to 30, and some of the package versions changed as well. This change is most likely transparent, but if your Android SDK installation does not support version 30, this might be a problem.

### TLDR
It would be good to review this code before I merge it to the main branch so that everyone is up to speed.
It would also be good to build and do a quick test to make sure everyone can still build/run the app.

### CODE REVIEW

**Comment by carbonXIII** [comment]
Oh, and a few features for User Story 1 might be missing, so lmk if anything stands out.
